zerver/lib/streams.py

```diff
@@ -408,6 +408,10 @@ def filter_stream_authorization(user_profile: UserProfile,
         if stream.id in streams_subscribed:
             continue
 
+        # Web public streams are accessible even to guests
+        if stream.is_web_public:
+            continue
+
         # Users are not authorized for invite_only streams, and guest
         # users are not authorized for any streams
```

Comment:
Let's update this comment and maybe change this next block in a preparatory refactoring to be:
```
# Members and administrators are authorized for public streams
if not stream.invite_only and not user_profile.is_guest:
    continue

unauthorized_streams.append(stream)
```

(I just think the flow will be better with all the cases being `continue`).